vminfo_parser/analyzer.py

```diff
@@ -492,57 +432,23 @@ def generate_unsupported_os_counts(self: t.Self) -> pd.Series:
-        self: t.Self, dataFrame: pd.DataFrame, os_name: str, minimum_count: int
-    ) -> pd.DataFrame:
-
-        filtered_df = dataFrame[(dataFrame["OS Name"] == os_name)]
-        counts = filtered_df["OS Version"].fillna("unknown").value_counts().reset_index()
+    def get_os_version_distribution(self: t.Self, os_name: str) -> pd.Series:
+        df_copy = self.vm_data.df.copy()
+        df_copy = df_copy[(df_copy["OS Name"] == os_name)]
+        counts = df_copy["OS Version"].fillna("unknown").value_counts().reset_index()
         counts.columns = ["OS Version", "Count"]
 
-        if minimum_count is not None and minimum_count > 0:
-            counts = counts[counts["Count"] >= minimum_count]
+        if self.config.minimum_count is not None and self.config.minimum_count > 0:
+            counts = counts[counts["Count"] >= self.config.minimum_count]
 
         return counts
 
-    def sort_attribute_by_environment(
-        self: t.Self,
-        *env_keywords: str,
-        attribute: str = "operatingSystem",
-        os_filter: t.Optional[str] = None,
-        environment_filter: t.Optional[str] = None,
-        show_disk_in_tb: bool = False,
-        over_under_tb: bool = False,
-        granular_disk_space_by_os: bool = False,
-    ) -> None:
-        env_column = "Environment"
-        data_cp = self.vm_data.df.copy()
-        if env_column not in self.vm_data.df.columns:
-            if "ent-env" in self.vm_data.df.columns:
-                env_column = "ent-env"
-            else:
-                raise ValueError("Neither 'Environment' nor 'ent-env' found in DataFrame columns.")
-
-        data_cp[env_column] = self.vm_data.df[env_column].apply(self.categorize_environment, args=env_keywords)
+    def by_os(self: t.Self, func: Callable[[str], None]) -> None:
+        """Execute func once for each os in get_unique_os_names.
 
-        if os_filter:
-            data_cp = data_cp[data_cp["OS Name"] == os_filter]
-
-        if environment_filter and environment_filter not in ["all", "both"]:
-            data_cp = data_cp[data_cp[env_column] == environment_filter]
-
-        if data_cp.empty:
-            LOGGER.warning("None found in %s", environment_filter)
-            return
-
-        if attribute == "diskSpace":
-            self.handle_disk_space(
-                data_cp,
-                environment_filter,
-                os_filter,
-                show_disk_in_tb=show_disk_in_tb,
-                over_under_tb=over_under_tb,
-                granular_disk_space_by_os=granular_disk_space_by_os,
-            )
-        if attribute == "operatingSystem":
-            self.handle_operating_system_counts(environment_filter, dataFrame=data_cp)
+            Func should have one argument (os_name), and return None
+        Args:
+            func (Callable[[str], None]): Function to execute for each os.
+        """
+        for os_name in self.get_unique_os_names():
+            func(os_name)
```

Comment:
not sure why this diff looks this way,  sort_attribute_by_environment was removed,  by_os is new